pkg/components/pluggable/discovery.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluggable
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/dapr/dapr/utils"
+	"github.com/dapr/kit/logger"
+
+	"github.com/pkg/errors"
+
+	"github.com/jhump/protoreflect/grpcreflect"
+
+	"google.golang.org/grpc"
+	reflectpb "google.golang.org/grpc/reflection/grpc_reflection_v1alpha"
+)
+
+var (
+	discoveryLog        = logger.NewLogger("pluggable-components-discovery")
+	onServiceDiscovered map[string]func(name string, dialer GRPCConnectionDialer)
+)
+
+func init() {
+	onServiceDiscovered = make(map[string]func(name string, dialer GRPCConnectionDialer))
+}
+
+// AddServiceV1DiscoveryCallback adds a callback function that should be called when the given service was discovered.
+func AddServiceV1DiscoveryCallback(serviceName string, callbackFunc func(name string, dialer GRPCConnectionDialer)) {
+	onServiceDiscovered[withV1(serviceName)] = callbackFunc
+}
+
+// removeExt removes file extension
+func removeExt(fileName string) string {
+	return fileName[:len(fileName)-len(filepath.Ext(fileName))]
+}
+
+const (
+	SocketFolderEnvVar  = "DAPR_PLUGGABLE_COMPONENTS_SOCKETS_FOLDER"
+	defaultSocketFolder = "/dapr-components-sockets"
```

Comment:
What about a better default, so people can use today without requiring a change in CLI?
`/tmp/dapr-components-sockets` This way we don't require people to use sudo for local development.

Reply:
fixed